src/v/storage/tests/compaction_e2e_test.cc

```diff
@@ -909,13 +907,14 @@ TEST_P(CompactionFixtureTombstonesParamTest, TestTombstonesCompletelyEmptyLog) {
         }
     }
 
-    // Sleep for tombstone.retention.ms time, so that the next time we attempt
-    // to compact the tombstone records will be eligible for deletion.
-    ss::sleep(tombstone_retention_ms).get();
+    // Sleep for a very short amount of time to ensure that tombstone records
+    // will be eligible for deletion during the next round of compaction
+    ss::sleep(100ms).get();
```

Comment:
Is there anything unique about this test that makes this more likely? If not, can we update other tests that follow this pattern of sleeping for `tomebstone_retention_ms`?

Reply:
This test has only flaked for the segment size = 1 case, so I assume the timing was just not permissive enough.

Other tests have a lot more work happening as well as restarts, etc in between compaction rounds. If any other tests begin to show signs of flaking I will update them.